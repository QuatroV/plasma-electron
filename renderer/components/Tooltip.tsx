import React from "react";
import Tippy from "@tippyjs/react/headless";

const Tooltip = ({ children, tooltip, ...other }) => (
  <Tippy
    delay={11500}
    render={(attrs) => (
      <div
        className="animate-scale rounded bg-gray-800 p-2 text-xs text-white shadow transition-all"
        tabIndex={-1}
        {...attrs}
      >
        {tooltip}
      </div>
    )}
    {...other}
  >
    <div className="">{children}</div>
  </Tippy>
);

export default Tooltip;
