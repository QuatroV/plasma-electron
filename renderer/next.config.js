module.exports = {
  webpack: (config, { isServer }) => {
    if (!isServer) {
      config.target = 'electron-renderer';
    }
    
    config.module.rules.push({
      test: /\.svg$/i,
      issuer: /\.[jt]sx?$/,
      use: ['@svgr/webpack'],
    })

    return config;
  },
};
