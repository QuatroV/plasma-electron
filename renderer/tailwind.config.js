/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './renderer/pages/**/*.{js,ts,jsx,tsx}',
    './renderer/components/**/*.{js,ts,jsx,tsx}',
    './renderer/features/**/*.{js,ts,jsx,tsx}',
  ],
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
      },
      keyframes: {
        "slow-appear": {
          '0%': { transform: 'scale(0.1, 1)', transform: 'translate(-5%, 0)', opacity: "0.1" },
          '100%': { transform: 'scale(1, 1)', transform: 'translate(0, 0)', opacity: "1" },
        },
        "slow-appear-no-opacity": {
          '0%': { transform: 'scale(0.1, 1)', transform: 'translate(-5%, 0)'},
          '100%': { transform: 'scale(1, 1)', transform: 'translate(0, 0)'},
        },
        "roll": {
          from: { transform: 'rotate(0deg)'},
          to: { transform: 'rotate(-360deg)'},
        },
        "shake": {
          "10%":{
            transform: "translate3d(-1px, 0, 0)",
          },
          
          "20%":{
            transform: "translate3d(2px, 0, 0)",
          },
        
          "30%":{
            transform: "translate3d(-4px, 0, 0)",
          },
        
          "40%":{
            transform: "translate3d(4px, 0, 0)",
          },

          "50%":{
            transform: "translate3d(-4px, 0, 0)",
          },
        
          "60%":{
            transform: "translate3d(4px, 0, 0)",
          },

          "70%":{
            transform: "translate3d(4px, 0, 0)",
          },

          "80%":{
            transform: "translate3d(2px, 0, 0)",
          },

          "90%":{
            transform: "translate3d(-1px, 0, 0)",
          },
          
        },
        "scale": {
          from: { transform: 'scale(0)'},
          to: { transform: 'scale(1)'},
        }
      },
      animation: {
        'slow-appear': 'slow-appear 0.25s linear',
        "slow-appear-no-opacity": 'slow-appear-no-opacity 0.25s linear',
        'one-roll': 'roll .25s linear',
        'fast-shake': 'shake .25s linear',
        'scale': 'scale .12s linear',
      },
    },
    fontFamily: {
      rubik: ['Rubik', 'sans-serif'],
    }
  },
  plugins:[
    require('@tailwindcss/container-queries'),
  ],
};
