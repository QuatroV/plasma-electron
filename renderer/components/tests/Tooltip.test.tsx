import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Tooltip from "../Tooltip";

describe("Spinner", () => {
  // Hack, see link: https://github.com/facebook/react/issues/11565#issuecomment-380143358
  beforeAll(() => {
    //@ts-ignore
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  });

  afterEach(() => {
    //@ts-ignore
    ReactDOM.createPortal.mockClear();
  });
  it("renders correctly", () => {
    const tree = renderer
      .create(<Tooltip tooltip="tooltip">Child</Tooltip>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
