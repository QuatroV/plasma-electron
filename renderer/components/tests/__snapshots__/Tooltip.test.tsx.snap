// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
[
  <div
    className=""
  >
    Child
  </div>,
  <div
    className="animate-scale rounded bg-gray-800 p-2 text-xs text-white shadow transition-all"
    tabIndex={-1}
  >
    tooltip
  </div>,
]
`;
